pkg/metrics/metrics_api_test.go

```diff
@@ -0,0 +1,237 @@
+package metrics
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/alicebob/miniredis/v2"
+	"github.com/inngest/inngest/pkg/consts"
+	"github.com/inngest/inngest/pkg/enums"
+	"github.com/inngest/inngest/pkg/execution/state/redis_state"
+	"github.com/inngest/inngest/pkg/publicerr"
+	"github.com/redis/rueidis"
+	"github.com/stretchr/testify/require"
+)
+
+// setupRedis creates a mock Redis instance for testing
+func setupRedis(t *testing.T) (redis_state.QueueManager, func()) {
+	r := miniredis.RunT(t)
+
+	rc, err := rueidis.NewClient(rueidis.ClientOption{
+		InitAddress:  []string{r.Addr()},
+		DisableCache: true,
+	})
+	require.NoError(t, err)
+
+	// Create queue shard with Redis client
+	defaultShard := redis_state.QueueShard{
+		Kind:        string(enums.QueueShardKindRedis),
+		RedisClient: redis_state.NewQueueClient(rc, redis_state.QueueDefaultKey),
+		Name:        consts.DefaultQueueShardName,
+	}
+
+	// Create queue manager (which implements QueueManager interface)
+	qm := redis_state.NewQueue(defaultShard)
+
+	return qm, func() {
+		rc.Close()
+	}
+}
+
+// testAuthMiddleware creates a simple test auth middleware
+func testAuthMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		authHeader := r.Header.Get("Authorization")
+		if authHeader == "" {
+			_ = publicerr.WriteHTTP(w, publicerr.Wrap(nil, http.StatusUnauthorized, "Authorization header required"))
+			return
+		}
+
+		if !strings.HasPrefix(authHeader, "Bearer ") {
+			_ = publicerr.WriteHTTP(w, publicerr.Wrap(nil, http.StatusUnauthorized, "Bearer token required"))
+			return
+		}
+
+		token := strings.TrimPrefix(authHeader, "Bearer ")
+		if token != "valid-test-token" {
+			_ = publicerr.WriteHTTP(w, publicerr.Wrap(nil, http.StatusUnauthorized, "Invalid token"))
+			return
+		}
+
+		// Token is valid, continue to next handler
+		next.ServeHTTP(w, r)
+	})
+}
+
+func TestMetricsAPIIntegration(t *testing.T) {
+	// Setup Redis mock
+	queueManager, cleanup := setupRedis(t)
+	defer cleanup()
+
+	// Setup metrics API with auth middleware
+	api, err := NewMetricsAPI(Opts{
+		AuthMiddleware: testAuthMiddleware,
+		QueueManager:   queueManager,
+	})
+	require.NoError(t, err)
+
+	tests := []struct {
+		name           string
+		authHeader     string
+		expectedStatus int
+		checkBody      func(t *testing.T, body string)
+	}{
+		{
+			name:           "authenticated request should return metrics",
+			authHeader:     "Bearer valid-test-token",
+			expectedStatus: http.StatusOK,
+			checkBody: func(t *testing.T, body string) {
+				// Should contain Prometheus format with queue_depth metric
+				require.Contains(t, body, "# HELP queue_depth Total system queue depth")
+				require.Contains(t, body, "# TYPE queue_depth gauge")
+				require.Contains(t, body, "queue_depth")
```

Comment:
convention wise, it should prefix with the service name so this should be
`inngest_queue_depth` instead